src/PIL/BlpImagePlugin.py

```diff
@@ -0,0 +1,418 @@
+        )
+        self.size = struct.unpack("<II", header.read(8))
+        offsets = struct.unpack("<16I", header.read(16 * 4))
+        lengths = struct.unpack("<16I", header.read(16 * 4))
+        palette_data = self.fp.read(256 * 4)
+
+        self.mode = "RGB"
+        self.tile = []
+        if compression == 1:  # Uncompressed or DirectX compression
+            data = []
+            self.fp.seek(offsets[0])
+            if encoding == 1:  # uncompressed
+                palette = getpalette(palette_data)
+                _data = BytesIO(self.fp.read(lengths[0]))
+                while True:
+                    try:
+                        offset, = struct.unpack("<B", _data.read(1))
+                    except struct.error:
+                        break
+                    b, g, r, a = palette[offset]
+                    data.append(struct.pack("<BBB", r, g, b))
+
+            elif encoding == 2:  # directx compression
+                if alpha_encoding == 0:  # DXT1
+                    linesize = int((self.size[0] + 3) / 4) * 8
+                    for yb in range(int((self.size[1] + 3) / 4)):
+                        line_data = self.fp.read(linesize)
+                        if alpha_depth:
+                            self.mode = "RGBA"
+                            decoded = decode_dxt1(line_data, alpha=True)
+                        else:
+                            decoded = decode_dxt1(line_data)
+                        for d in decoded:
+                            data.append(d)
+
+                elif alpha_encoding == 1:  # DXT3
+                    linesize = int((self.size[0] + 3) / 4) * 16
+                    self.mode = "RGBA"
+                    for yb in range(int((self.size[1] + 3) / 4)):
+                        decoded = decode_dxt3(self.fp.read(linesize))
+                        for d in decoded:
+                            data.append(d)
+
+                elif alpha_encoding == 7:  # DXT5
+                    linesize = int((self.size[0] + 3) / 4) * 16
+                    self.mode = "RGBA"
+                    for yb in range(int((self.size[1] + 3) / 4)):
+                        decoded = decode_dxt5(self.fp.read(linesize))
+                        for d in decoded:
+                            data.append(d)
+                else:
+                    raise NotImplementedError(
+                        "Alpha encoding %r not supported" % (alpha_encoding)
+                    )
+
+            self.im = Image.core.new(self.mode, self.size)
+
+            data = b"".join(data)
+            self.frombytes(data)
+
+    def _open(self):
+        magic = self.fp.read(4)
+        self.fp.seek(0)
+        if magic == b"BLP1":
+            return self._decode_blp1()
+
+        if magic == b"BLP2":
+            return self._decode_blp2()
+
+        raise ValueError("not a BLP file (magic: %r)" % (magic))
+
+
+Image.register_open("BLP", BlpImageFile)
+Image.register_extension("BLP", ".blp")
```

Comment:
Hi. Would you be interested in replacing `"BLP"` with `BlpImageFile.format` in these two lines, to avoid duplication, as per #1333?

Reply:
Done